libs/auth/src/java_card.ts

```diff
@@ -576,15 +570,17 @@ export class JavaCard implements Card {
     );
   }
 
-  private getData(objectId: Buffer): Promise<Buffer> {
-    return this.cardReader.transmit(
+  private async getData(objectId: Buffer): Promise<Buffer> {
+    const dataTlv = await this.cardReader.transmit(
       new CardCommand({
         ins: GET_DATA.INS,
         p1: GET_DATA.P1,
         p2: GET_DATA.P2,
         data: constructTlv(GET_DATA.TAG_LIST_TAG, objectId),
       })
     );
+    const [, , data] = parseTlv(PUT_DATA.DATA_TAG, dataTlv);
+    return data;
```

Comment:
This makes it such that there's parity between `putData` and `getData`. The data provided to `putData` is now exactly what you get back from `getData`.